### DATA CHECK NOTES

Forgive the mess! This repo has a lot of extra files right now. Here's what I've been up to, though:

I'm rearranging this so all the ETL is in standalone script files -- both .py and .R ("0-3_etl*") and the analysis is in analysis.Rmd file.

I also did some transforms so that all of the manual work would be in one step. The etl files ultimately output: [data/processed/2_etl_combined.xlsx](data/processed/2_etl_combined.xlsx). The ones that were coded as pertaining to pregnant patients are in [data/manual/confirmed_pregnant.xlsx](data/manual/confirmed_pregnant.xlsx), and the ones that were not are in [data/manual/not_confirmed_pregnant.xlsx](data/manual/not_confirmed_pregnant.xlsx) -- mostly generated from the following. Not sure where to put this code yet, should maybe leave it unpublished and only publish the resultant data sets.

In [49]:
import pandas

etl_2_combined = pandas.read_excel("data/processed/2_etl_combined.xlsx")
manual_pregnant_patient = pandas.read_excel("data/processed/EMTALA_PREGNANT_2011-2022.xlsx")
more_to_remove = ["2409 PTBH11","2409 D6YK11","2409 IT9611","2407 74BC11","2409 VRM911","2409 NZOV11","2406 05FK11","2407 ZY3N11","2406 JOU711","2406 4F4G11","2406 8E2Y11","2409 PKRO11","2407 T0DE11","2406 CITS12","2406 S4NX11","2406 RU2V11","2407 UD9X11","2407 KZUZ11","2406 D9FE11"]

pregnant_patient_keys = manual_pregnant_patient["key_identifier"].str.strip().unique().tolist()
not_confirmed_pregnant = etl_2_combined[~(etl_2_combined["key_identifier"].str.strip().isin(pregnant_patient_keys))|(etl_2_combined["key_identifier"].str.strip().isin(more_to_remove))|(etl_2_combined["hospital_type"]=="Psychiatric")]
not_confirmed_pregnant.to_excel("data/manual/not_confirmed_pregnant.xlsx",index=False)
print(not_confirmed_pregnant.shape)

confirmed_pregnant = etl_2_combined[(etl_2_combined["key_identifier"].str.strip().isin(pregnant_patient_keys))&(~etl_2_combined["key_identifier"].str.strip().isin(more_to_remove))&(etl_2_combined["hospital_type"]!="Psychiatric")]

turnaway = pandas.read_excel("data/processed/turnaway.xlsx")
turnaway["turnaway"] = turnaway["turnaway"].str.strip()
turnaway_keys = turnaway[turnaway["turnaway"].apply(lambda x: str(x)[0]=="Y")]["key_identifier"].unique()
confirmed_pregnant["turnaway"] = confirmed_pregnant["key_identifier"].isin(turnaway_keys)

confirmed_pregnant.to_excel("data/manual/confirmed_pregnant.xlsx",index=False)

(135, 19)


/var/folders/n1/qf74t1dj555cgs6f_mhz9nm0m6xrwl/T/ipykernel_11910/2161310210.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  confirmed_pregnant["turnaway"] = confirmed_pregnant["key_identifier"].isin(turnaway_keys)


In [51]:
confirmed_pregnant

Unnamed: 0 key_identifier  method   
1             1    2405 XQIL11    both  \
2             2    2406 XQIL11    both   
3             3    2406 GLJ011    both   
4             4    2406 75KL11    both   
5             5    2407 75KL11    both   
..          ...            ...     ...   
802         646    2409 KYYH11  simple   
803         651    2406 VS8U11  simple   
804         652    2406 CGE011  simple   
806         654    2407 QIQV11  simple   
808         664    2402 M9UO11  simple   

                              facility_name              hospital_type   
1        EUREKA SPRINGS HOSPITAL COMMISSION  Critical Access Hospitals  \
2        EUREKA SPRINGS HOSPITAL COMMISSION  Critical Access Hospitals   
3                 MERCY HOSPITAL BERRYVILLE  Critical Access Hospitals   
4                      GLENN MEDICAL CENTER  Critical Access Hospitals   
5                      GLENN MEDICAL CENTER  Critical Access Hospitals   
..                                      ...                        ...   
802  HOUSTON METHODIST WILLOWBROOK HOSPITAL                 Short Term   
803           PROVIDENCE CENTRALIA HOSPITAL                 Short Term   
804         GRAYS HARBOR COMMUNITY HOSPITAL                 Short Term   
806                                   TRIOS                 Short Term   
808                MERCY HEALTH SYSTEM CORP                 Short Term   

     facility_id                  address            city state   
1          41304         24 NORRIS STREET  EUREKA SPRINGS    AR  \
2          41304         24 NORRIS STREET  EUREKA SPRINGS    AR   
3          41329        214 CARTER STREET      BERRYVILLE    AR   
4          51306       1133 W SYCAMORE ST         WILLOWS    CA   
5          51306       1133 W SYCAMORE ST         WILLOWS    CA   
..           ...                      ...             ...   ...   
802       450844  18220 STATE HIGHWAY 249         HOUSTON    TX   
803       500019      914 S SCHEUBER ROAD       CENTRALIA    WA   
804       500031       915 ANDERSON DRIVE        ABERDEEN    WA   
806       500053  900 SOUTH AUBURN STREET       KENNEWICK    WA   
808       520066   1000 MINERAL POINT AVE      JANESVILLE    WI   

     deficiency_tag  missing_survey_tag_count             dfcncy_desc defpref   
1              2405                       NaN      EMERGENCY ROOM LOG       C  \
2              2406                       NaN  MEDICAL SCREENING EXAM       C   
3              2406                       NaN  MEDICAL SCREENING EXAM       C   
4              2406                       NaN  MEDICAL SCREENING EXAM       C   
5              2407                       NaN   STABILIZING TREATMENT       C   
..              ...                       ...                     ...     ...   
802            2409                       NaN    APPROPRIATE TRANSFER     NaN   
803            2406                       NaN  MEDICAL SCREENING EXAM     NaN   
804            2406                       NaN  MEDICAL SCREENING EXAM     NaN   
806            2407                       NaN   STABILIZING TREATMENT     NaN   
808            2402                       NaN        POSTING OF SIGNS     NaN   

    inspection_date EVENT_ID   
1        2013-11-14   XQIL11  \
2        2013-11-14   XQIL11   
3        2016-08-02   GLJ011   
4        2013-10-10   75KL11   
5        2013-10-10   75KL11   
..              ...      ...   
802      2015-12-18   KYYH11   
803      2016-04-28   VS8U11   
804      2016-01-06   CGE011   
806      2022-03-25   QIQV11   
808      2016-07-20   M9UO11   

                                       inspection_text   
1     _x000d_\n_x000d_\nBased on review of the emer...  \
2     _x000d_\n_x000d_\nBased on review of Medical ...   
3     _x000d_\n_x000d_\nBased on interview and lack...   
4    **NOTE- TERMS IN BRACKETS HAVE BEEN EDITED TO ...   
5    **NOTE- TERMS IN BRACKETS HAVE BEEN EDITED TO ...   
..                                                 ...   
802  _x000D__x000D_\n_x000D__x000D_\nBased

I did spot checks of both of these files. I didn't find anything in the "confirmed_pregnant" file that pertained to non-pregnant patients, but I did find a few in the "not_confirmed_pregnant" file that appeared to pertain to pregnant patients.

#### I think there's a handful of cases that we're excluding that we might want to include -- log and signage violations

In [42]:
import pandas 

spotcheck = pandas.read_excel("data/manual/not_confirmed_pregnant--spotcheck.xlsx",index_col=0)
spotcheck[spotcheck["pregnant?"]=="yes"][["key_identifier","notes"]]

key_identifier                                              notes
Unnamed: 0                                                                  
68            2402 YOYJ11  it's a signage thing that definitely affected ...
660           2402 ZSG311  it's a signage thing that definitely affected ...
523           2405 78J911  it's a log keeping thing that definitely affec...
169           2402 WEKB11  it's a signage thing that definitely affected ...
517           2405 724D11  it's a log keeping thing that definitely affec...
447           2405 X8E611  it's a log keeping thing that definitely affec...
522           2402 78J911  it's a signage policy thing that definitely af...
11            2402 C4HB11  it's a signage thing that definitely affected ...

#### There are a handful of "Y" responses that we're not counting among the turnaways

Most of them look like the turned-away patient was not actually pregnant, but at least two (pre-eclampsia, flagging as good insight) look like they were?

In [41]:
turnaway = pandas.read_excel("data/processed/turnaway.xlsx")
turnaway["turnaway"] = turnaway["turnaway"].str.strip()
turnaway[turnaway["turnaway"].apply(lambda x: str(x)[0]=="Y")].groupby("turnaway")["EVENT_ID"].nunique().sort_index()


turnaway
Y                                                        237
Y -- flagging as good insight into action by hospital      1
Y -- motor vehicle accident                                1
Y -- pre-eclampsia                                         1
Y -- same case same pregnant person                        1
Y -- same patient                                          1
Name: EVENT_ID, dtype: int64

#### I'm definitely not following anything in 3_etl_hosp_summary.R

Tried to clean up what was going on in R, but am not good enough at R to make heads or tails of it.

When I do what I think should be happening in python, I'm getting different counts (Rachel found 135 rural, 203 non-rural)! The findings still hold true.

Web: “Scripps News manually reviewed hundreds of CMS investigations involving pregnant patients of both rural and urban hospitals.”

Script: “OUR INVESTIGATION FOUND THE CASES INVOLVE BOTH SMALL, RURAL FACILITIES..AND THOSE IN BIG CITIES”


In [80]:
rurban_hosp = pandas.read_excel("data/public/rural_urban_hospitals.xlsx")
rurban_hosp[rurban_hosp["facility_id"].duplicated()]
confirmed_pregnant["facility_id"] = confirmed_pregnant["facility_id"].astype(str)

hosp_summ = confirmed_pregnant.groupby(["facility_id","facility_name","city","state"]).agg({"EVENT_ID":"nunique","dfcncy_desc":list_unique,"inspection_date":list_unique}).reset_index()
emtala_hosp_rurban = hosp_summ.join(rurban_hosp.set_index("facility_id"),on="facility_id",rsuffix="_rurban")

inspections_by_rurban = emtala_hosp_rurban.groupby("Rural Status")["EVENT_ID"].sum()
hospitals_by_rurban = emtala_hosp_rurban.groupby("Rural Status")["facility_id"].nunique()
print("Total number of inspections finding violations in the time period (unique EVENT_ID), broken down by whether or not the hospital was rural:",inspections_by_rurban)
print("Total number of hospitals in violation in the time period (unique facility_id), broken down by whether or not the hospital was rural:",hospitals_by_rurban)




Total number of inspections finding violations in the time period (unique EVENT_ID), broken down by whether or not the hospital was rural: Rural Status
No     179
Yes    137
Name: EVENT_ID, dtype: int64
Total number of hospitals in violation in the time period (unique facility_id), broken down by whether or not the hospital was rural: Rural Status
No     165
Yes    128
Name: facility_id, dtype: int64


/var/folders/n1/qf74t1dj555cgs6f_mhz9nm0m6xrwl/T/ipykernel_11910/761598602.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  confirmed_pregnant["facility_id"] = confirmed_pregnant["facility_id"].astype(str)
